examples/signals/src/lib.rs

```diff
@@ -0,0 +1,123 @@
+use gdnative::*;
+
+#[derive(NativeClass)]
+#[inherit(Node)]
+// register_with attribute can be used to specify custom register function for node signals and properties
+#[register_with(Self::register_signals)]
+struct SignalEmitter {
+    timer: f64,
+    data: i64,
+}
+
+#[methods]
+impl SignalEmitter {
+    fn register_signals(builder: &init::ClassBuilder<Self>) {
+        builder.add_signal(init::Signal {
+            name: "something_happens",
```

Comment:
Nitpick: I'd prefer shorter names for the signals, for example "tick" and "tock", since we're triggering signals every second, and not assigning any specific meaning to them.

Reply:
Long singal names are better for educational pup

Reply:
Long names are better for educational purposes of the tutorial. Also, imho, they give some insight about normal signal naming. Better explicit than "which baboom it is" half a year after:)

Reply:
Long names are not better -- descriptive names are. In terms of information conveyed to the users, "something_happens" is even less informative when compared to "tick": the latter indicates that the signal will be emitted periodically while the former doesn't.

It doesn't have to be my suggestions: if you have any ideas for more descriptive names, we can go for those as well, but "something_happens" is far too generic a phrase to use here.

Reply:
Nah. Changed to tick and tick_with_data